plugins/communicators/ssh/communicator.rb

```diff
@@ -400,7 +400,16 @@ def connect(**opts)
         raise Vagrant::Errors::SSHNotReady if ssh_info.nil?
 
         # Default some options
-        opts[:retries] = 5 if !opts.key?(:retries)
+        opts[:retries] = if ssh_info.key?(:retries)
```

Comment:
the `ssh_info` should probably already have the default value for the retries (coming from the config). So, there is no need to default this again. Instead, this can be  `opts[:retries] = ssh_info[:retries] if !opts.key?(:retries)` (same for the retry_interval below). This also implies that the internal caller of this method can override the user defined config. 

This also means that the [`ssh_info` method in the `Machine`](https://github.com/hashicorp/vagrant/blob/620df8fd0fdb74d087f051feba988e4131248f3b/lib/vagrant/machine.rb#L448) class needs to be updated to account for this new config.